js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -1028,11 +1028,11 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
         this._$selectAllItem = $('<div>').addClass(SELECT_ALL_ITEM_CLASS);
 
         const value = this._dataAdapter.isAllSelected();
-        this._createComponent(this._$selectAllItem, CheckBox, {
+        const checkBox = this._createComponent(this._$selectAllItem, CheckBox, {
             value: value,
-            text: this.option('selectAllText'),
-            onValueChanged: this._onSelectAllCheckboxValueChanged.bind(this)
+            text: this.option('selectAllText')
         });
+        checkBox.on('valueChanged', this._onSelectAllCheckboxValueChanged.bind(this));
```

Comment:
It's not clear why we have to change to this approach.

Reply:
Replaced with old approach with using of the option method